person.rb

```diff
@@ -11,6 +11,7 @@ def initialize(age, name = 'unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rental = []
   end
```

Comment:
- [ ] Kindly create a method `add_rental` in the `Book` class that takes in `person`, `book`, and `date` as parameters. The method should create a new `Rental`. Remember that `Person` and `Book` have a _**many to many**_ relationship through `Rental`. This should be mirrored in the `Book` class. You can read this [article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) to get a better understanding. 👍

Reply:
I think what I understood from the example is that `person` and `book` has `many to many` relationships and when a a new `rental` object is created it at the same time we will add the `rental` to `@rental` of the `person` and `book`. That's we added` person.rental << self` and `book.rental << self` as part of the constructor

Reply:
Hi @awaisanwar544, you should be able to access the `Rental` class from the `Book` and the `Person`. That is the reason why you have to create the `add_rental` method in the `Book` and `Person`. That way you can create a new Rental from the `Book` and `Person` classes. If you are not satisfied with the review and you feel it should be approved, you can suggest a second opinion and a senior code reviewer will have a look at it. 👍

Reply:
@Gambit142 I'm trying to understand how this relationship between different classes works. I really appreciate that you have answered my questions. Although I still didn't understand the concept properly I'll do the required changes. Maybe there should have been more reading material and a guide for this.